src/pocketmine/Player.php

```diff
@@ -2295,18 +2296,19 @@ public function sendForm(Form $form) : void{
 	 * @return bool
 	 */
 	public function onFormSubmit(int $formId, $responseData) : bool{
-		if(!isset($this->forms[$formId])){
+		if(!isset($this->formHandlers[$formId])){
 			$this->logger->debug("Got unexpected response for form $formId");
 			return false;
 		}
 
 		try{
-			$this->forms[$formId]->handleResponse($this, $responseData);
+			$closure = $this->formHandlers[$formId];
+			$closure($this, $responseData);
 		}catch(FormValidationException $e){
-			$this->logger->critical("Failed to validate form " . get_class($this->forms[$formId]) . ": " . $e->getMessage());
+			$this->logger->critical("Failed to validate form: " . $e->getMessage());
```

Comment:
might be better to use `Utils::getNiceClosureName()` here for debug info instead of removing it completely.